validator/client/runner_test.go

```diff
@@ -0,0 +1,132 @@
+package client
+
+import (
+	"context"
+	"testing"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+)
+
+func cancelledContext() context.Context {
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	return ctx
+}
+
+func TestRun_initializesValidator(t *testing.T) {
+	v := &fakeValidator{}
+	run(cancelledContext(), v)
+	if !v.InitializeCalled {
+		t.Error("Expected Initialize() to be called")
+	}
+}
+
+func TestRun_cleansUpValidator(t *testing.T) {
+	v := &fakeValidator{}
+	run(cancelledContext(), v)
+	if !v.DoneCalled {
+		t.Error("Expected Done() to be called")
+	}
+}
+
+func TestRun_waitsForActivation(t *testing.T) {
+	v := &fakeValidator{}
+	run(cancelledContext(), v)
+	if !v.WaitForActivationCalled {
+		t.Error("Expected WaitForActivation() to be called")
+	}
+}
+
+func TestRun_onNextSlot_updatesAssignments(t *testing.T) {
+	v := &fakeValidator{}
+	ctx, cancel := context.WithCancel(context.Background())
+
+	slot := uint64(55)
+	ticker := make(chan uint64)
+	v.NextSlotRet = ticker
+	go func() {
+		ticker <- slot
+
+		cancel()
+	}()
+
```

Comment:
empty lines

Reply:
These are purposeful grouping of setup logic vs the method under test.